src/core/qgsproject.cpp

```diff
@@ -358,7 +358,7 @@ QgsProject::QgsProject( QObject *parent )
   , mLayerStore( new QgsMapLayerStore( this ) )
   , mBadLayerHandler( new QgsProjectBadLayerHandler() )
   , mSnappingConfig( this )
-  , mRelationManager( new QgsRelationManager( this ) )
+  , mRelationManager( new QgsRelationManager( *this ) )
```

Comment:
I'd stick to the other managers convention and pass a pointer. We should probably move all of them to unique ptrs eventually but for now I'd recommend to be consistent with the other calls..